add external secrets profile with crs kustomization reference (#202)
weaveworks/profiles-catalog: charts/external-secrets/Bootstrapping-Guide.md

# [Guide][How-To] Secrets Management with flux

# Goal

Bootstrapping external secret operator with flux

## Installation problem

In Flux, we can't have dependencies between Flux Kustomization and HelmRelease, so we install `external-secrets-operator` through a `HelmRelease` and the `Secrets` CRs (`SecretStore`, `ExternalSecret, ..`) through a Flux Kustomization.

> Both controllers manage the resources independently, at different moments, with no possibility to wait each other. This means that we have a wonderful race condition where sometimes the CRs (`SecretStore`,`ClusterSecretStore`...) tries to be deployed before than the CRDs needed to recognize them.

Reviewer: ```suggestion
Both controllers manage the resources independently, at different moments, with no possibility to wait for each other. This means that we have a wonderful race condition where sometimes the CRs (`SecretStore`,`ClusterSecretStore`...) tries to be deployed before than the CRDs needed to recognize them.
```

  sourceRef:
    kind: GitRepository
    name: flux-system
  path: ./secrets
  prune: true
  validation: client
```

- ***clusters/my-cluster/cluster-secrets***

This is the Kustomization file, the manifest of external secrets resources

**Contents:**

```yaml
apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
kind: Kustomization
metadata:
  name: cluster-secrets
  namespace: flux-system
spec:
  interval: 5m
  sourceRef:
    kind: GitRepository
    name: flux-system
  path: ../cluster-secrets
  prune: true
  validation: client
```

- ***secrets/aws-secret-store.yaml***

This file has the configuration for aws-authentication using key and secret, or you can make it using service account

```yaml
apiVersion: external-secrets.io/v1beta1
kind: SecretStore
metadata:
  name: aws-secret-store
  namespace: flux-system
spec:
  provider:
    aws:
      service: SecretsManager
      region: eu-north-1
      auth:
        secretRef:
          accessKeyIDSecretRef:
            name: awssm-secret
            key: access-key
          secretAccessKeySecretRef:
            name: awssm-secret
            key: secret-access-key
```

- ***secrets/aws.yaml***

> This CR has describes what data should be fetched, how the data should be transformed and saved as a `Kind=Secret`

Reviewer: ```suggestion
This CR describes which secret should be fetched from the secret store, how the secret should be transformed and saved as a `Kind=Secret`
```
